crates/core/src/config.rs

```diff
@@ -1,18 +1,37 @@
 use std::path::Path;
 use std::{fmt, io};
 
+use toml;
+use toml_edit::{DocumentMut, Formatted, Item, Table, Value};
+
 use spacetimedb_lib::Address;
 use spacetimedb_paths::cli::{ConfigDir, PrivKeyPath, PubKeyPath};
 use spacetimedb_paths::server::{ConfigToml, MetadataTomlPath};
 
-#[derive(serde::Serialize, serde::Deserialize)]
-pub struct MetadataFile {
-    pub version: semver::Version,
-    pub edition: String,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub client_address: Option<Address>,
+pub fn current_version() -> semver::Version {
+    env!("CARGO_PKG_VERSION").parse().unwrap()
+}
+
+/// Parse a TOML file at the given path, returning `None` if the file does not exist.
+///
+/// **NOTE**: Comments and formatting in the file could be preserved.
+pub fn parse_preserving_config<T: for<'a> From<&'a DocumentMut>>(
```

Comment:
This is a silly name. Of course parsing the config preserves the config. Do you want to call this `parse_config_preserving_formatting` or something?

Reply:
Also, why is this given such a general type? Can we not just have a function that parses a `RawConfig`? Or, failing that, why is this not just a function that reads a `DocumentMut` from a file, and then the caller gets to parse it into whatever `T` they want on their own?

Reply:
Also, if this is only going to be called by the CLI, then it should be in the CLI's config.rs, not here.

Reply:
> This is a silly name. Of course, parsing the config preserves the config. Do you want to call this `parse_config_preserving_formatting` or something?

This is to make a difference with `parse_config` that uses the regular `serde` deserialization that doesn't preserve the formatting.

Reply:
> Also, why is this given such a general type? Can we not just have a function that parses a `RawConfig`? Or, failing that, why is this not just a function that reads a `DocumentMut` from a file, and then the caller gets to parse it into whatever `T` they want on their own?

I do this in case other files need preserving, and because `RawConfig` is not accessible here.

Then, because `parse_config` is here too, move this out will hide later the fact we have 2 ways to read it.